pkg/lib/cockpit-components-terminal.tsx

```diff
@@ -261,19 +376,20 @@ export class Terminal extends React.Component {
     }
 
     componentWillUnmount() {
-        this.disconnectChannel();
-        this.terminal.dispose();
         window.removeEventListener('resize', this.onWindowResize);
         this.onFocusOut();
+        this.unmountTerminal(this.terminal_state);
+        if (!this.props.state)
+            this.terminal_state.close();
```

Comment:
This added line is not executed by any test.  [Details](https://logs-cockpit.apps.ocp.cloud.ci.centos.org/pull-21680-b536158f-20250901-105310-fedora-42-devel/Coverage/lcov/github-pr.diff.gcov.html)